xbmc/addons/AddonManager.cpp

```diff
@@ -805,7 +805,8 @@ bool CAddonMgr::CanAddonBeDisabled(const std::string& ID)
     return false;
 
   CSingleLock lock(m_critSection);
-  if (IsSystemAddon(ID))
+  // Non-optional system add-ons can not be disabled
+  if (IsSystemAddon(ID) && !IsOptionalSystemAddon(ID))
```

Comment:
Do we really need both these checks? I'm not sure a system addon can be in both lists.

Reply:
NVM, system addons include the optional ones.

Reply:
An optional system add-on is still a system add-on. And should be treated the same except for it can be disabled. That's why there are two lists.